### PR TITLE
10.0 mig l10n nl partner salutation port from 8.0 to 10.0

### PR DESCRIPTION
**Pull request by daramousk**

### CODE REVIEW

**Comment by NL66278** [comment]
This PR contains all the stuff from the 109 PR. Understandable as you need PR 109 for this one to function. In this case it might be better to close 109 and use 110 as a PR that migrates the dutch name-system to Odoo 10.0. I made some remarks on coding standards on the 109 PR that applies to both modules migrated in this.

**Comment by NL66278** [comment]
And you might want some tests for the salutation functionality.

File: l10n_nl_partner_salutation/README.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Module Dutch Partner Name
```
Comment (NL66278) [inline l10n_nl_partner_salutation/README.rst]: Dutch Partner Salutation

File: l10n_nl_partner_salutation/README.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+While organizations have no gender (the field is hidden there), you can have
+a feminine salutation by creating a partner title with an implicit gender,
+such as 'Ladies club'.
+
+Contributors
+------------
+
+* Holger Brunn <hbrunn@therp.nl>
+* Ronald Portier <ronald@therp.nl>
+* Stefan Rijnhart <stefan@therp.nl>
```
Comment (NL66278) [inline l10n_nl_partner_salutation/README.rst:28]: And add your name here

File: l10n_nl_partner_salutation/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+##############################################################################
```
Comment (NL66278) [inline l10n_nl_partner_salutation/__openerp__.py]: Short headers needed also here

File: l10n_nl_partner_salutation/model/res_partner.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    )
+    salutation_address = fields.Char(
+        compute='_get_salutation',
+        string='Salutation (address)'
+    )
+    use_manual_salutations = fields.Boolean('Enter salutations manually')
+    salutation_manual = fields.Char('Manual salutation (letter)')
+    salutation_address_manual = fields.Char('Manual salutation (address)')
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (NL66278) [inline l10n_nl_partner_salutation/model/res_partner.py]: And no vim line here

**Comment by daramousk** [comment]
@NL66278 Changes pushed

**Review by NL66278 — APPROVED** [review]
LGTM

**Review by hbrunn — CHANGES_REQUESTED** [review]
concerning the code, that's fine. But you need to pull the commits from 8.0 as described in https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-10.0#technical-method-to-migrate-a-module-from-90-to-100-branch

Just do this in a new branch, cherry-pick your commits on it and push --force to overwrite this branch. Then also squash all your commits so that we have just one commit that does the migration

File: l10n_nl_partner_name/model/res_partner.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+
+class ResPartner(models.Model):
+    """Extend res.partner with extra fields for Dutch names."""
+    _inherit = 'res.partner'
+
+    initials = fields.Char()
+    infix = fields.Char()
+
+    @api.one
```
Comment (espo-tony) [inline l10n_nl_partner_name/model/res_partner.py]: `@api.one` is deprecated. Please, use `@api.multi` instead and either iterate over the recordset or use `ensure_one()`

File: l10n_nl_partner_salutation/model/res_partner.py [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    @api.onchange('use_manual_salutations')
+    def on_change_use_manual_salutations(self):
+        """Handle onchange event."""
+        if self.use_manual_salutations:
+            # When changing to manual salutation, initially
+            # use the computed fields for default.
+            self.salutation_manual = self.salutation
+            self.salutation_address_manual = self.salutation_address
+
+    def get_salutation_name_format(self):
```
Comment (espo-tony) [inline l10n_nl_partner_salutation/model/res_partner.py]: Better decorate this method with `@api.model`

**Review by espo-tony — CHANGES_REQUESTED** [review]

**Review by espo-tony — APPROVED** [review]

**Comment by hbrunn** [comment]
@daramousk if you follow up on https://github.com/OCA/l10n-netherlands/pull/110#pullrequestreview-57470858 and squash your commits, I can merge this

**Comment by daramousk** [comment]
@hbrunn Done

**Comment by hbrunn** [comment]
is is possible you missed some commits when cherry picking? CI is red again...

**Comment by NL66278** [comment]
Red runbot was due to runbot failure. After Force Rebuild the following warning remains:
2017-09-13 09:40:08,015 187 WARNING openerp_test odoo.addons.base.ir.ir_model: Creating the ir.model.data res_partner_title_sir in module base instead of l10n_nl_partner_salutation.

Is there a reason to put the title in base?

**Comment by daramousk** [comment]
@NL66278 As I understand, this is happening because we want to override the original records in `addons/base/res/res_partner_data.xml`.

**Comment by daramousk** [comment]
@hbrunn ping

**Review by hbrunn — CHANGES_REQUESTED** [review]
nice. Now say `git rebase -i 10.0` and squash as follows:

```
pick 8769e337 [ADD] l10n_nl_partner_name, l10n_nl_partner_salutation
pick f6d1df82 [FIX] show empty string if lastname is unset
pick a4127e43 [UPD] make it work with latest partner-contact
pick 94dd99e2 [FIX] flake8
pick db484e9f [ADD] tests
pick db89e257 [UPD] prefix versions with 8.0
pick 53a7c34c [FIX] linting
pick edb0763e OCA Transbot updated translations from Transifex
pick 6f8ec0b1 [MIG] Migrates l10n_nl_partner_name from 8.0 to 10.0
s cf715189 [MIG] Migrates l10n_nl_partner_salutation from 8.0 to 10.0
s 1df9f1ab Adding a commiter, changing the format of the licence comment, removing deprecated Char init variable
# pick 1907a990b27faf655bd9820eaa4c9e99aef31d8c Merge branch '10.0-mig-l10n_nl_partner_name' into 10.0-mig-l10n_nl_partner_salutation
s 210a4867 [FIX] Sort version of licencing, removing VIM comments, changing __openerp__ to __manifest__
s cb6f99e3 [FIX] Adding tests to increase code coverage
s 4b2453e1 [FIX] Replaces api.one with api.multi and inserting api.model
s 46c17b34 [FIX] Adds newlines on the xmls, adds licence on manifest, changes name of compute methods
s beb6af73 [FIX] Creates the res_partner_title_sir in the module instead on the base module
```

**Comment by daramousk** [comment]
@hbrunn Squashed

**Review by hbrunn — APPROVED** [review]

**Pull request merged**